src/nci_core.c

```diff
@@ -894,16 +934,20 @@ nci_core_add_current_state_changed_handler(
     NciCoreFunc func,
     void* data)
 {
-    return nci_core_add_signal_handler(core, SIGNAL_CURRENT_STATE, func, data);
+    return nci_core_add_signal_handler(core, SIGNAL_CURRENT_STATE, 0,
+        G_CALLBACK(nci_core_closure_cb),
+        G_CALLBACK(func), data);
```

Comment:
Other add handler functions had data parameter renamed to user_data but not in this one.

Reply:
fixed, thanks!